client/src/App.tsx

```diff
@@ -23,18 +25,21 @@ function App(): JSX.Element {
         <SnackBarProvider>
           <AuthProvider>
             <SocketProvider>
-              <Switch>
-                <Route exact path="/signin" component={Login} />
-                <Route exact path="/signup" component={Signup} />
-                <Route path="/booking" component={Booking} />
-                <Route path="/dashboard" component={Dashboard} />
-                <Route exact path="/profile-details" component={ProfileDetails} />
-                <Route exact path="/myprofile" component={MyProfile} />
-                <Route exact path="/listing" component={Listing} />
-                <Route exact path="/sitting" component={Sitting} />
-                <Route exact path="/" component={Home} />
-                <Route path="*" component={NotFound} />
-              </Switch>
+              <ConversationProvider>
```

Comment:
It is not ideal to have the conversation provider wrap around the entire app, you can instead wrap only the components that need it. Let's for example, we are browsing listings and we receive a new message, it would cause the entire app to re-render 😄

Reply:
done ✔️